### PR TITLE
feat(native-filters): Adjust filter components for horizontal mode

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR adjusts native filter components for horizontal filter bar.
Noteworthy changes:
1. Value filter:
   a) Introduce a `oneLine` mode to Select component - if enabled, only 1 tag will be displayed at once and the rest of selected tags will be collapsed to an overflow tag. When user opens the dropdown, only the overflow tag is displayed. All tags are always displayed in a single line.
   b) If tag is truncated, display a tooltip with full name
   c) Customize `maxTagPlaceholder`, so that the overflow tag doesn't display 3 dots at the end, which takes precious space 
       (e.g. instead of the default `+ 1 ...` we show `+1` when 1 item is overflowing)
3. Range filter:
   a) Adjust the top and bottom margin to make sure that the slider is centered in horizontal mode
   b) Decrease font weight and line height so that the marks fit well in the horizontal filter bar
4. Time range filter: add styling to make sure that the pill is centered in horizontal mode

Known issues:
1. The time range pill's width is dynamic depending on selected time range ("No filter" pill is smaller than "2022-01-01 : 2022-12-01" pill). However, the filter component itself always has a fixed width, which means that if the pill is short, there might be some empty space between the time range filter and the filter next to it. We are planning to replace it soon with a new component, stay tuned!

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screenshot 2022-11-30 at 14 49 21

### TESTING INSTRUCTIONS
1. Enable `HORIZONTAL_FILTER_BAR` ff
3. Set horizontal mode for the native filters bar
5. Add various types of filters
6. Make sure that nothing breaks

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @codyml for review

### CODE REVIEW

**Review by geido — COMMENTED** [review]
Just two nits from a first look at the code

**Comment by kgabryje** [comment]
@michael-s-molina @geido @codyml I implemented the Select changes based on our yesterday discussion. Can you take a look?

File: superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx [hunk 1]
```diff
@@ -72,6 +83,60 @@ function reducer(
...
+    text-overflow: ellipsis;
+  }
+`;
+
+export interface CustomTagProps extends HTMLSpanElement {
+  label: ReactNode;
+}
+
+// TODO: use antd Tag props instead of any. Currently it's causing a typescript error
+const Tag = (props: any) => {
```
Comment (michael-s-molina) [inline superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx:104]: It seems the custom render is necessary to make `oneLine` work correctly so we need to move this code to the Select component.

**Review by michael-s-molina — COMMENTED** [review]
Thanks for the PR @kgabryje. I left a couple of comments.

File: superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx [hunk 2]
```diff
@@ -323,11 +325,23 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
           onChange={handleChange}
           ref={inputRef}
           loading={isRefreshing}
-          maxTagCount={5}
+          maxTagCount={
+            filterBarOrientation === FilterBarOrientation.HORIZONTAL
+              ? 'responsive'
+              : 5
```
[the remark this answers is not in the document]
Comment (kgabryje) [reply superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx]: No longer applicable

File: superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx]: done

File: superset-frontend/src/components/Select/styles.tsx [hunk 3]
```diff
@@ -52,6 +52,25 @@ export const StyledSelect = styled(AntdSelect, {
...
+        flex-shrink: 1;
+        min-width: ${theme.gridUnit * 13}px;
+      }
+
+      .ant-select-selection-overflow-item-suffix {
+        flex: unset;
+        min-width: 0px;
+      }
+    `
+    }
```
[the remark this answers is not in the document]
Comment (kgabryje) [reply superset-frontend/src/components/Select/styles.tsx]: done

**Review by codyml — APPROVED** [review]
LGTM!  One thing I noticed that can probably wait for a future PR (or left as-is) is that if there's a time filter in the overflow dropdown, clicking either of the popover buttons also closes the dropdown, which is probably not what we want:

https://user-images.githubusercontent.com/13007381/205151199-ff098950-8b8e-4940-adac-f67f29b0efbc.mov

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thanks for the PR!

@kgabryje Let's open a follow-up to add the same behavior for the async version 😉

**Comment by michael-s-molina** [comment]
> LGTM! One thing I noticed that can probably wait for a future PR (or left as-is) is that if there's a time filter in the overflow dropdown, clicking either of the popover buttons also closes the dropdown, which is probably not what we want:

Good point @codyml. Let's handle that in a follow-up.

**Pull request merged**